staging_vespalib/src/vespa/vespalib/metrics/ticker_thread.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+#include "ticker_thread.h"
+#include "simple_metrics_manager.h"
+#include <chrono>
+
+#include <vespa/log/log.h>
+LOG_SETUP(".vespalib.metrics.ticker_thread");
+
+namespace vespalib {
+namespace metrics {
+
+void
+TickerThread::doTickerLoop(TickerThread *me)
+{
+    me->tickerLoop();
+}
+
+void
+TickerThread::tickerLoop()
+{
+    const std::chrono::seconds oneSec{1};
+    std::unique_lock<std::mutex> locker(_lock);
+    while (_runFlag) {
+        auto r = _cond.wait_for(locker, oneSec);
+        if (r == std::cv_status::timeout) {
+            _owner->tick();
+        }
+    }
+}
+
+void
+TickerThread::stop()
+{
+    std::unique_lock<std::mutex> locker(_lock);
+    _runFlag.store(false);
```

Comment:
Technically doesn't have to be an `atomic` since all accesses to it are covered by `_lock`, but no harm in having it.